### PR TITLE
fix(balance): repair chance rebalancing

### PR DESCRIPTION
**Pull request by chaosvolt**
## Purpose of change

It was pointed out in https://github.com/cataclysmbnteam/Cataclysm-BN/pull/3720 that you need literally superhuman tailoring skill to be able to reinforce uncraftable items (if they have no `repairs_like` defined), specifically level 16.

## Describe the solution

C++ changes:
1. In iuse_actor.cpp, changed how `repair_item_actor::repair_chance` determines `action_difficulty` when reinforcing, from favoring `recipe_difficulty` to instead being 50% harder than the value used for refitting, which in turn is set to the difficulty of repairing a max-damage item. Since max damage would return a value of 4, this makes the `action_difficulty` always 6, instead of being anywhere from 4 to 10 depending on how easy it is to craft (when `recipe_difficulty` is already used later on so how easy it is to craft doesn't need to be taken into account twice over).
2. In the same function, changed how `tool_quality` is used to determine `success_chance` and `damage_chance`. Before 1 point of tool quality was worth one fifth a point in skill level. Now it's used as a multiplier on skill level, with a value of 0 meaning no impact on skill, and a quality of 10 doubling it. The aim here was to make tool quality more relevant instead of barely mattering at all, enough that at max level it's worth a full level's worth each. Making it a multiplier instead of just increasing the original bonus fivefold was so that it doesn't disproportionately impact repair difficulty at lower levels.

JSON changes:
1. Changed how the tool bonuses for tailoring tools scale. Instead of only ever being a range of -1 to +1, the wooden needle has a bonus of zero, the bone and curved needles (idea being bone needle is more work to obtain typically and finer point) has a bonus of 1, sewing kit has a bonus of 2, and tailor's kit has a bonus of 3. All other repair tools had a range of 4-10 for their tool qualities, so it stands out as odd that tailoring tools were so much vastly worse.

Doc updates:
1. Documented the basics of the `repair_item` use action in json_info.md

All these factors combined should mean:
* Repairs at level zero skill will be slightly harder, since your tool quality bonus is being multiplied against zero. We're talking less than a level's worth of difference however since the old fixed tool quality bonus was so low.
* Once you gain any level of skill, tool quality starts to have a more noticeable impact on repairs, some items potentially being in reach a bit easier than they used to be. Given how long repairs can take, it should be a reasonable change.
* Once you max out your skill, you'll be just in range to successfully reinforce an uncraftable item with the right tools. In the case of tailoring, makeshift sewing tools will still be inadequate, but a sewing kit or tailor's kit will do the job, while all the more advanced metalworking tools will be much more reliable at repairing anything.

## Describe alternatives you've considered

Adding code that can check if an item would be literally impossible to refit with the current tool even at 10 repair skill, setting it to no longer count as a valid repair skill on the item's inventory screen, and making it say "this item is too hard for any human to reinforce, go get an exodii-made sewing kit" or something to that effect on trying to select such an item to reinforce.

## Testing

1. Checked affected JSON file for syntax and lint errors.
2. Compiled and load-tested.
3. Spawned in a moisture retention suit (an item I can currently confirm has maxed difficulty to repair) and debugged tailoring to 10.
4. Confirmed I can refit it with a bone sewing needle, but can't reinforce.
5. Confirmed a sewing kit bumps things up enough that I have a shot at reinforcing it.
6. Checked affected C++ file for astyle.

### CODE REVIEW

File: src/iuse_actor.cpp [hunk 1]
```diff
@@ -3315,9 +3315,11 @@ std::pair<float, float> repair_item_actor::repair_chance(
     // Duster |    2   |   5   |  5  |   10%   |   0%
     // Duster |    2   |   2   |  10 |   4%    |   1%
     // Duster | Refit  |   2   |  10 |   0%    |   N/A
```
Comment (scarf005) [inline src/iuse_actor.cpp:3317]: is this table updated?

**Review by scarf005 — APPROVED** [review]

File: src/iuse_actor.cpp (hunk 1 above)
Comment (chaosvolt) [reply src/iuse_actor.cpp:3317]: I don't think this table's been up to date for a while given the introduction of repairs like. It's not very readable either to be honest so I'm not sure whether to try and replace it or, if not, what to replace it with.

Comment (scarf005) [reply src/iuse_actor.cpp:3317]: if it's unmaintained would be great to remove it, since incorrect comments are often more dangerous than no comments.

Comment (chaosvolt) [reply src/iuse_actor.cpp]: And done :>

**Review by scarf005 — APPROVED** [review]

**Comment by chaosvolt** [comment]
Noticed it was commenting on the multiplication applied to the reinforce math, fixed that since target value can also be obtained by slapping +2 onto it.

**Review by scarf005 — APPROVED** [review]

**Pull request merged**